nixos/doc/manual/administration/cleaning-store.xml

```diff
@@ -14,14 +14,14 @@ collector</emphasis> to remove old, unreferenced packages.  This is
 easy:
 
 <screen>
-$ nix-collect-garbage
+# nix-collect-garbage
```

Comment:
There are mentions of it. The main description is on http://nixos.org/nix/manual/#ssec-multi-user. It's what allows unprivileged users using `nix-env`.